src/sentry/sentry_apps/models/sentry_app.py

```diff
@@ -34,25 +35,72 @@
 from sentry.types.region import find_all_region_names, find_regions_for_sentry_app
 from sentry.utils import metrics
 
+
+class SentryAppResourceType(StrEnum):
+    ISSUE = "issue"
+    ERROR = "error"
+    COMMENT = "comment"
+    INSTALLATION = "installation"
+    METRIC_ALERT = "metric_alert"
+
+    # Represents an issue alert resource
+    EVENT_ALERT = "event_alert"
+
+
+class SentryAppActionType(StrEnum):
+    # Generic actions
+    CREATED = "created"
+    DELETED = "deleted"
+    UPDATED = "updated"
+
+    # Issue actions
+    ASSIGNED = "assigned"
+    IGNORED = "ignored"
+    RESOLVED = "resolved"
+    UNRESOLVED = "unresolved"
+
+    # Issue alert actions
+    TRIGGERED = "triggered"
+
+    # Metric alert actions
+    OPEN = "open"
+    CRITICAL = "critical"
+    WARNING = "warning"
+
+
+# Define event type unions for each resource
+IssueEventType = Literal[
+    "issue.assigned", "issue.created", "issue.ignored", "issue.resolved", "issue.unresolved"
+]
+ErrorEventType = Literal["error.created"]
+CommentEventType = Literal["comment.created", "comment.deleted", "comment.updated"]
+
+# Union of all event types
+EventType = Union[IssueEventType, ErrorEventType, CommentEventType]
+
 # When a developer selects to receive "<Resource> Webhooks" it really means
 # listening to a list of specific events. This is a mapping of what those
 # specific events are for each resource.
-EVENT_EXPANSION = {
-    "issue": [
+EVENT_EXPANSION: Final[dict[str, list[EventType]]] = {
+    SentryAppResourceType.ISSUE.value: [
```

Comment:
nit: this is a `StrEnum` so you don't have to use `.value` to get the `str`

```suggestion
    SentryAppResourceType.ISSUE: [
```